cirq-google/cirq_google/optimizers/two_qubit_gates/gate_compilation.py

```diff
@@ -1,22 +1,16 @@
 # pylint: disable=wrong-or-nonexistent-copyright-notice
 """Attempt to tabulate single qubit gates required to generate a target 2Q gate
-with a product A k A."""
-from functools import reduce
-from typing import Tuple, Sequence, List, NamedTuple
+with a product A k A.
+
+Attention: this module is now deprecated! Use the classes from cirq-core instead."""
+from typing import Tuple, Sequence, NamedTuple
 
 from dataclasses import dataclass
 import numpy as np
 
 import cirq
-from cirq import value
-from cirq._compat import proper_repr, proper_eq
-from cirq_google.optimizers.two_qubit_gates.math_utils import (
-    kak_vector_infidelity,
-    vector_kron,
-    weyl_chamber_mesh,
-    random_qubit_unitary,
-    kak_vector_to_unitary,
-)
+from cirq import TwoQubitGateTabulation, two_qubit_gate_product_tabulation
+from cirq._compat import deprecated, deprecated_class
 
 _SingleQubitGatePair = Tuple[np.ndarray, np.ndarray]
 
```

Comment:
Also deprecate `TwoQubitGateCompilation` class in favour of cirq's `TwoQubitGateTabulationResult`.

Reply:
I think this might not be necessary: that class will go away as long as the old method is deprecated. It is not exported anywhere. Deprecating it in favour of the new class would require exporting that class - if there is a need for that, could it be done as a follow-up?

Reply:
Update: added the export and deprecation anyway (without serialization of the exported class).